carting/views.py

```diff
@@ -7,33 +11,95 @@
+
+    sections = (
+        OuvrageSection.objects.filter(geometry__bboverlaps=bbox)
+        # FIXME : A geometry is not excluded by filter below 75f7dbbd-c0b4-456f-bc10-19f72f89608b
+        .exclude(geometry__contains_properly=bbox).filter(
+            typology__in=[x.name for x in allowed_typologies]
+        )
+    )
+
+    geojson = serialize("geojson", (s for s in sections if s.geometry))
+
+    return render(
+        request,
+        "carting/search_by_position.html",
+        {
+            "scroll_snap": True,
+            "geojson": geojson,
+            "bbox": json.dumps(qs_bbox),
+            "zoom_level": zoom_level,
+            "sections": sections,
+        },
+    )
+
+
+@require_GET
+def search_by_position_details(
+    request: HttpRequest, root_expanded: UUID, expanded: UUID
+) -> HttpResponse:
+    zoom_level = float(request.GET.get("zoom", 1))
+
+    qs_bbox = request.GET.get("bbox", SAINT_MALO_BBOX)
+    qs_bbox = [float(coordinate) for coordinate in qs_bbox.split(",")]
+
+    expanded_section = OuvrageSection.objects.prefetch_related("children").get(
+        pk=expanded
+    )
+
+    to_serialize = [expanded_section, *expanded_section.children.all()]
+    geojson = serialize("geojson", (s for s in to_serialize if s.geometry))
+
+    return render(
+        request,
+        "carting/search_by_position.html",
+        {
+            "scroll_snap": True,
+            "geojson": geojson,
+            "bbox": json.dumps(qs_bbox),
+            "zoom_level": zoom_level,
+            "root_expanded": root_expanded,
+            "expanded": expanded,
+            "expanded_section": expanded_section,
+        },
+    )
+
 
 # FIXME : Les sections commençant par '0.' ne devraient pas être affichées (pas de géométrie attachée); les illustrations en '0.' sont mal ordonnées
 @require_GET
-def index(request: HttpRequest) -> HttpResponse:
+def search_by_text(request: HttpRequest) -> HttpResponse:
     search = request.GET.get("search", "")
 
     if not search:
-        return redirect(reverse("carting:index") + "?search=4.1.")
+        return redirect(reverse("carting:search_by_text") + "?search=4.1.")
 
     if not search.endswith("."):
-        return redirect(reverse("carting:index") + f"?search={search}.")
+        return redirect(reverse("carting:search_by_text") + f"?search={search}.")
 
     ouvrage, _, numero = search.rpartition("/")
     if not numero:
         return render(
             request,
-            "carting/index.html",
+            "carting/search_by_text.html",
         )
-
-    sections = OuvrageSection.objects.exclude(
-        typology__in=[
-            SectionTypology.ALINEA.name,
-            SectionTypology.ILLUSTRATION.name,
-            SectionTypology.REFERENCE.name,
-            SectionTypology.TABLE.name,
-            SectionTypology.TOPONYME.name,
-        ]
+    allowed_typologies = SectionTypology.paragraph_likes | {SectionTypology.OUVRAGE}
```

Comment:
I am not 100% sure to understand this line. Is there a situation where `SectionTypology.paragraph_likes` is falsy ?

Reply:
https://docs.python.org/3/library/stdtypes.html#frozenset.union